src/andromede/simulation/linear_expression.py

```diff
@@ -132,15 +68,23 @@ def _str_for_coeff(coeff: float) -> str:
         return "{:+g}".format(coeff)
 
 
-def _str_for_time_expansion(exp: TimeExpansion) -> str:
-    if isinstance(exp, TimeShiftExpansion):
-        return f".shift({exp.shift})"
-    elif isinstance(exp, TimeSumExpansion):
-        return f".sum({exp.from_shift}, {exp.to_shift})"
-    elif isinstance(exp, AllTimeExpansion):
-        return ".sum()"
-    else:
-        return ""
+def _time_index_to_str(time_index: TimeIndex) -> str:
+    if isinstance(time_index, TimeShift):
+        if time_index.timeshift == 0:
+            return "t"
+        elif time_index.timeshift > 0:
+            return f"t + {time_index.timeshift}"
+        else:
+            return f"t - {-time_index.timeshift}"
+    if isinstance(time_index, TimeStep):
+        return f"{time_index.timestep}"
+    return ""
+
+
+def _scenario_index_to_str(scenario_index: ScenarioIndex) -> str:
```

Comment:
Could we print the current scenario `w` for `NoScenarioIndex` instances ?

Reply:
Indeed, and NoScenarioIndex is not a good naming actually, I think we should distinguish `CurrentScenarioIndex` and `NoScenarioIndex` for things that are not scenario-dependent.

Reply:
I improved the implementation to distinguish between "CurrentScenario" and "NoScenario":
NoScenario is for parameters and variables which are not scenario-dependent, CurrentScenario means the scenario of the constraint we are building, for scenario-dependent parameters and variables.